src/models/GitHub.js

```diff
@@ -54,22 +57,27 @@ class GitHub extends Fetcher {
       date.setDate(date.getDate() - 31)
     }
     const dateStr = date.toISOString()
-    return this.get(`notifications?since=${encodeURIComponent(dateStr)}`,
-                    { fetchAll: true }).then(result => result.json)
+    const url = `notifications?since=${encodeURIComponent(dateStr)}`
+    return this.get(url).then(result => result.json)
   }
 
   // https://developer.github.com/v3/search/#search-issues
-  getTasks(query) {
-    const params = `?q=${encodeURIComponent(query)}`
-    const url = `${Config.githubApiUrl}/search/issues${params}`
+  getTasks(filter) {
+    const extraParams = filter.updatedAt ? ` updated:>${filter.updatedAt}` : ' is:open'
+    const params = `?q=${encodeURIComponent(filter.query + extraParams)}`
+    const url = `${Config.githubApiUrl}/search/issues${params}&per_page=100`
     return this.getTasksFromUrl(url)
   }
 
-  getTasksFromUrl(url) {
-    return this.get(url).then(result => {
-      const { json, nextUrl } = result
-      return { tasks: json.items.map(d => getTask(d)), nextUrl,
-               currentUrl: url }
+  getTasksFromUrl(url, items = []) {
+    return this.get(url).then(({ json, headers }) => {
+      const allItems = items.concat(json.items)
+      const nextUrl = this.getNextUrl(headers)
+      if (nextUrl) {
+        return this.getTasksFromUrl(nextUrl, allItems)
```

Comment:
Fetches all tasks, 100 tasks per page. But because subsequent fetches will use the `updated:>${filter.updatedAt}` filter, I don't think it will auto-paginate very often.